<launch>
	<!-- Get point clouds -->
  <node pkg="point_cloud_assembler" type="laser_scan_assembler_srv" output="screen"
           name="laser_scan_assembler">

           <remap from="scan_in" to="tilt_scan"/>
           <param name="tf_cache_time_secs" type="double" value="10.0" />
	   <param name="max_scans" type="int" value="400" />
    	   <param name="ignore_laser_skew" type="bool" value="true" />
	   <param name="fixed_frame" type="string" value="odom" />
   </node>

   <node pkg="point_cloud_assembler" type="point_cloud_snapshotter" output="screen" name="snapshotter">
        <remap from="laser_scanner_signal" to="laser_tilt_controller/laser_scanner_signal"/>
	<remap from="build_cloud" to="laser_scan_assembler/build_cloud" />
	<remap from="full_cloud" to="snapshot_cloud" />
	<param name="fixed_frame" type="string" value="odom" />
   </node>

   <!-- Get planar patch maps -->
   <node pkg="planar_patch_map" type="planar_patch_map_node" >
       <remap from="/tilt_laser_cloud" to="/snapshot_cloud"/>
   </node>


     <!-- A converter from planar_patch_map to a blank annotated map -->
     <node pkg="annotated_planar_patch_map" type="empty_annotated_map" 
     	   name="patch_map_converter" respawn="false" output="screen">

	   <remap from="scan_in" to="/planar_map"/>
	   <param name="fixed_frame" value="odom"/>


     </node>	   




     <node pkg="annotated_planar_patch_map" type="annotated_map_assembler_srv" 
     	   name="blank_map_server" respawn="false" 
	   output="screen">

	   <remap from="/poly_object_map" to="/empty_poly_map"/>
	   <param name="fixed_frame" value="odom"/>

	   <param name="tf_tolerance_secs" value="0.0"/>
	   <param name="tf_cache_time_secs" value="500.0"/>
     </node>	   



     <node pkg="annotated_planar_patch_map" type="annotated_map_snapshotter"  name="am_snapshotter" respawn="false" output="screen">
        <remap from="build_annotated_map" to="/blank_map_server/build_map" />
        <remap from="full_map" to="/labeled_maps/blank" />
   
     </node>




</launch>
